spring-core/src/main/java/org/springframework/core/convert/support/NumberToNumberConverterFactory.java

```diff
@@ -40,11 +43,18 @@
  * @see java.math.BigDecimal
  * @see NumberUtils
  */
+@SuppressWarnings({"rawtypes", "unchecked"})
 final class NumberToNumberConverterFactory implements ConverterFactory<Number, Number>, ConditionalConverter {
+	private final static Map<Class<?>, NumberToNumber<?>> cache = new ConcurrentHashMap<>();
```

Comment:
upper-case; java naming convention

Reply:
the thing is, when you add these `Class<?>` to a static cache - you make them GC roots, forever, in this case. Unless there is serious proof that this allocation is perf critical, this might harm more then fix.

Reply:
Thanks, use Class<?> getName as a key.

Reply:
that's... not what I meant. I mean the fact that you declare a `private static final Map... ` means that you declare an additional GC root. In general, you should not think about these details and just code the way it is easier to read.

Let's change the angle of this PR a bit, may be that will make more sense. _Why_ is there a need to fix _anything_ here? Because on each invocation there will be a new instance allocated, right? And let me ask you then - why is that bad? Or is it bad at all?

Reply:
Yes, we have an instance allocated on each invocation but also we call ConversionUtils.getEnumType(targetType) for each invocation. When we use static cache we save small amount of time on this operations, but we need some warm up the server to work it the fastest way. Also, I want notice that ConversionUtils.getEnum() have a cycle inside. Also, we lost some memory 'cause of cache. I'l be glad if you say some words about how harmful GC root can be.

So, I want just to save some time inconversions by this pr.

Reply:
- I was not saying that the addition of this Map is _harmful_; all I said is that I am not convinced this is something that needs to be "fixed". The allocation of such a small Object looks like a very tiny optimization (especially considering that the world is moving towards concurrent GCs, like `Shenandoah` or `ZGC` - where these allocations will not even matter).

- do you have any proof that you are saving some time, actually? and of course even if that is true, you are paying with memory for that. But now that you add a GC root - you are also _adding_ time for the scanning itself (though it will be very, very little; and for most GCs it will be concurrent).

 - overall, as I said initially, this has to be coded in manner that is easier to read and not think (much) about the details above. I thought that may be you have done some experiments and have some numbers, thus the initial questions.